test/e2e/storage/testsuites/provisioning.go

```diff
@@ -1358,8 +1355,60 @@ func findVolumeMountPath(pod *v1.Pod, claim *v1.PersistentVolumeClaim) string {
 	return containerMountPath
 }
 
-// getFilesystemSizeBytes returns a total size of a filesystem on given mountPath inside a pod. You can use findVolumeMountPath for mountPath lookup.
-func getFilesystemSizeBytes(pod *v1.Pod, mountPath string) (int, error) {
+// getFilesystemSizeBytes returns a total size of a filesystem on given mountPath inside a pod.
+// It detects the node's operating system (Windows or Linux) and executes the appropriate command
+// inside the target pod to determine the total capacity of the mounted volume's filesystem.
+// You can use findVolumeMountPath for mountPath lookup.
+func getFilesystemSizeBytes(pod *v1.Pod, mountPath string) (int64, error) {
+
+	if framework.NodeOSDistroIs("windows") {
+		// --- WINDOWS Implementation: Use fsutil volume diskfree ---
+
+		// 1. Extract Drive Letter: Windows paths (mountPath) are expected to start with a drive letter,
+		// e.g., "C:\var\lib\kubelet\...". fsutil requires only the drive prefix (e.g., "C:").
+		if len(mountPath) < 2 || mountPath[1] != ':' {
+			return 0, fmt.Errorf("invalid Windows mount path: %q", mountPath)
+		}
+		drive := mountPath[:2] // e.g., "C:"
+
+		// 2. Execute fsutil Command: Run fsutil inside the pod to get disk usage statistics.
+		// We use cmd.exe /c to ensure the command executes correctly in the container environment.
+		out, err := e2ekubectl.RunKubectl(
+			pod.Namespace,
+			"exec", pod.Name, "--",
+			"cmd.exe", "/c",
+			fmt.Sprintf("fsutil volume diskfree %s", drive),
+		)
+		if err != nil {
+			return 0, fmt.Errorf("failed running fsutil: %w", err)
+		}
+
+		// 3. Parse Output: fsutil output uses Windows line endings (\r\n) and a specific format.
+		// Normalize line endings before splitting into lines.
+		// Output structure:
+		// Total # of bytes             : <total_bytes>
+		lines := strings.Split(strings.ReplaceAll(out, "\r\n", "\n"), "\n")
+		if len(lines) < 2 {
+			return 0, fmt.Errorf("unexpected fsutil output: %q", out)
+		}
+
+		// The total size is on the second line (index 1).
+		// Use SplitN to isolate the value after the first colon, handling potential formatting variation.
+		parts := strings.SplitN(lines[1], ":", 2)
+		if len(parts) != 2 {
+			return 0, fmt.Errorf("unable to parse total bytes line: %q", lines[1])
+		}
+
+		// 4. Convert to int64: Extract the byte value and parse it as a 64-bit integer.
+		val := strings.TrimSpace(parts[1])
+		totalBytes, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return 0, fmt.Errorf("parsing total bytes: %w", err)
+		}
+
+		return totalBytes, nil
```

Comment:
Please split it into `getFilesystemSizeBytesWindows` and `getFilesystemSizeBytesLinux` to keep the functions small and focused. `getFilesystemSizeBytes` will just have something like this

```go

	if framework.NodeOSDistroIs("windows") {
        return getFilesystemSizeBytesWindows(...)
    }
    return getFilesystemSizeBytesLinux(...)
```